### PR TITLE
GEN-44: Enable schedule and automerge for various dev deps

### PR DESCRIPTION
**Pull request by nonparibus**
## 🌟 What is the purpose of this PR?

Follows #2887 and enables automerge for various dev dependencies (linters and test tools).

## 🔗 Related links

- Renovate [docs on Automerge](https://docs.renovatebot.com/key-concepts/automerge)
- Renovate [docs on Scheduling](https://docs.renovatebot.com/key-concepts/scheduling/)
- Later.js [docs on text-based schedule definitions](https://breejs.github.io/later/parsers.html#text)

## 🚫 Blocked by

- https://github.com/hashintel/hash/pull/2887

## 🔍 What does this change?

- Modifies the `.github/renovate.json` file in line with the above.
- Additionally prevents the raising of updates for npm updates less than three days old (which may be at risk of unpublishing).

### 🚢 Has this modified a publishable library?

- [x] This PR does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

- [x] The changes in this PR are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

- [x] The changes in this PR do not affect the execution graph

## ⚠️ Known issues

- There are no known issues, but we should monitor whether enabling `lockFileMaintenance` automerge throws up any issues, in particular.
 
## 🐾 Next steps

- As we move towards enabling automerge more widely, we expect that certain changes in process may be required.
- We should assess whether enabling automerge in general has been worth it after a month of use, and monitor/address any issues that arise out of this PR.
- Attempt to enable `lockFileMaintenance`, removed in https://github.com/hashintel/hash/commit/22cbb2a7b5093867f522beb5678fef6f36243293

## 🛡 What tests cover this?

_n/a_

## ❓ How to test this?

_n/a_

## 📹 Demo

_n/a_

### CODE REVIEW

File: .github/renovate.json [hunk 1]
```diff
@@ -7,12 +7,22 @@
...
+  "npm": {"minimumReleaseAge": "3 days"},
   "postUpdateOptions": ["yarnDedupeFewer"],
   "rangeStrategy": "bump",
   "rebaseWhen": "conflicted",
   "semanticCommits": "disabled",
   "updateInternalDeps": "true",
 
+  "lockFileMaintenance": {
+    "enabled": true,
+    "automerge": true
```
Comment (TimDiekmann) [inline .github/renovate.json:19]: We have automerge enabled in line `5` as this requires an approval anyway.

File: .github/renovate.json [hunk 2]
```diff
@@ -7,12 +7,22 @@
...
   "ignorePaths": ["apps/engine/**"],
+  "npm": {"minimumReleaseAge": "3 days"},
   "postUpdateOptions": ["yarnDedupeFewer"],
   "rangeStrategy": "bump",
   "rebaseWhen": "conflicted",
   "semanticCommits": "disabled",
   "updateInternalDeps": "true",
 
+  "lockFileMaintenance": {
+    "enabled": true,
```
Comment (TimDiekmann) [inline .github/renovate.json:18]: We have automerge enabled in line `5` as this requires an approval anyway, so these two lines are the only real change in this PR.

File: .github/renovate.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
     },
     {
```
Comment (TimDiekmann) [inline .github/renovate.json:34]: automerge can be enabled here

Comment (nonparibus) [reply .github/renovate.json:33]: Realized the same, at about the same time :P https://github.com/hashintel/hash/pull/2892/commits/ab68632bab9a12b03ade53d23d4c50573cde7b38

File: .github/renovate.json [hunk 4]
```diff
@@ -21,7 +20,7 @@
...
       "enabled": true,
-      "groupName": "Block Protocol",
+      "groupName": "Block Protocol npm packages",
```
Comment (TimDiekmann) [inline .github/renovate.json:23]: Not worth investigating now, but having this automated would be nice at some point.

**Review by TimDiekmann — APPROVED** [review]

File: .github/renovate.json (hunk 4 above)
Comment (nonparibus) [reply .github/renovate.json:23]: I suspect it's doable. Renovate is **so** configurable. But it also **_so_** configurable I don't have the patience to wade through its voluminous docs right now. 😅

**Pull request merged**